From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <heftig@archlinux.org>
Date: Sun, 28 Sep 2025 18:09:42 +0200
Subject: [PATCH] Stop overwriting prepared image data

As mentioned in the bugs referenced, clearing the pixbuf in the prepared
signal is not a valid thing to do.

See: https://gitlab.gnome.org/GNOME/gdk-pixbuf/-/commit/75ac8deaf04f362efa5fc0ace51031350e70e8be
---
 src/image-load.cc | 23 +----------------------
 1 file changed, 1 insertion(+), 22 deletions(-)

diff --git a/src/image-load.cc b/src/image-load.cc
index 770f1ea69465..fdbe93fbf355 100644
--- a/src/image-load.cc
+++ b/src/image-load.cc
@@ -535,29 +535,8 @@ static void image_loader_area_updated_cb(gpointer,
 static void image_loader_area_prepared_cb(gpointer, gpointer data)
 {
 	auto il = static_cast<ImageLoader *>(data);
-	GdkPixbuf *pb;
-	guchar *pix;
-	size_t h;
-	size_t rs;
 
-	/* a workaround for
-	   https://bugzilla.gnome.org/show_bug.cgi?id=547669
-	   https://bugzilla.gnome.org/show_bug.cgi?id=589334
-	*/
-	g_autofree gchar *format = il->backend->get_format_name();
-	if (strcmp(format, "svg") == 0 ||
-	    strcmp(format, "xpm") == 0)
-		{
-		return;
-		}
-
-	pb = il->backend->get_pixbuf();
-
-	h = gdk_pixbuf_get_height(pb);
-	rs = gdk_pixbuf_get_rowstride(pb);
-	pix = gdk_pixbuf_get_pixels(pb);
-
-	memset(pix, 0, rs * h); /*this should be faster than pixbuf_fill */
+	(void) il;
 }
 
 static void image_loader_size_cb(gpointer,
